lib/foreman_maintain/reporter/cli_reporter.rb

```diff
@@ -208,8 +210,10 @@ def filter_decision(answer)
       # rubocop:disable Metrics/MethodLength
       def ask_to_select(message, steps)
         if assumeyes?
-          puts('(assuming first option)')
-          return steps.first
+          step = steps[@select_option_counter]
+          @select_option_counter += 1
```

Comment:
With this change we will end up on option 3 if somehow we fail to delete the task ? or some N no of tasks went to paused state while we are deleting previous tasks ?

Reply:
So should we enable maintenance mode while deleting the tasks? Enabling maintenance mode will take care that other task will not go in paused state.

Reply:
I think better way would be to conditionally have only two procedures passed to :next_steps in NotPaused check ? As example,
~~~
      def next_procedures
        if assumeyes?
          return  [Procedures::ForemanTasks::Resume.new,
                   Procedures::ForemanTasks::Delete.new(:state => :paused)]
        end
        [Procedures::ForemanTasks::Resume.new,
         Procedures::ForemanTasks::Delete.new(:state => :paused),
         Procedures::ForemanTasks::UiInvestigate.new('search_query' => scoped_search_query)]
      end

      assert(paused_tasks_count == 0,
             "There are currently #{paused_tasks_count} paused tasks in the system",
             :next_steps => next_procedures)
~~~